admin-dev/themes/new-theme/js/pages/product/edit/combinations-manager.ts

```diff
@@ -40,11 +41,45 @@ const CombinationEvents = ProductEventMap.combinations;
 const CombinationsMap = ProductMap.combinations;
 
 export default class CombinationsManager {
+  productId: number;
+
+  eventEmitter: typeof EventEmitter;
+
+  $productForm: JQuery;
+
+  $combinationsContainer: JQuery;
+
+  combinationIdInputsSelector: string;
+
+  $externalCombinationTab: JQuery;
+
+  $preloader: JQuery;
+
+  $paginatedList: JQuery;
+
+  $emptyState: JQuery;
+
+  paginator: DynamicPaginator | null;
+
+  combinationsRenderer: CombinationsGridRenderer | null;
```

Comment:
I'm not sure about these ones it should be instead:
```suggestion
  combinationsRenderer!: CombinationsGridRenderer;
```

Because the component is initialized no matter what in one of the innit functions, I'm also not sure it is useful to init these fields to null in the constructor Because nowhere do we check if `combinationsRenderer === null`

At least for this field, maybe for other fields it is necessary to handle nullable values, but I think it would be worth to check each one and fix them correctly